src/library/Uno.Themes.WinUI.Markup/Theme/ToggleSwitch.cs

```diff
@@ -0,0 +1,264 @@
+﻿using Microsoft.UI.Xaml;
+using Microsoft.UI.Xaml.Media;
+using Uno.Extensions.Markup.Internals;
+
+namespace Uno.Themes.Markup
+{
+	public static partial class Theme
+	{
+		public static class ToggleSwitch
+		{
+			public static class Resources
+			{
+				public static class Knob
+				{
+					public static class Stroke
+					{
+						[ResourceKeyDefinition(typeof(Brush), "SwitchKnobStrokeThickness")]
+						public static ResourceValue<Brush> Thickness => new("SwitchKnobStrokeThickness", true);
+					}
+
+					public static class Shadow
+					{
+						public static class Size
+						{
+							[ResourceKeyDefinition(typeof(Brush), "SwitchKnobShadowSize")]
+							public static ResourceValue<Brush> Default => new("SwitchKnobShadowSize", true);
+						}
+					}
+
+					public static class Icon
+					{
+						public static class Size
+						{
+							[ResourceKeyDefinition(typeof(Brush), "KnobIconSize")]
+							public static ResourceValue<Brush> Default => new("KnobIconSize", true);
+						}
+
+						[ResourceKeyDefinition(typeof(Thickness), "KnobIconPadding")]
+						public static ResourceValue<Thickness> Padding => new("KnobIconPadding", true);
+					}
```

Comment:
We don't need nested classes for these inside of Knob, they can just be straight properties like:

```csharp
[ResourceKeyDefinition(typeof(Thickness), "SwitchKnobStrokeThickness")]
public static ResourceValue<Thickness> StrokeThickness => new("SwitchKnobStrokeThickness", true);
```

Also, be careful to match the `typeof` to be the right type for the resource. Here you have the typeof as `Brush` when it should be `Thickness`

Reply:
resolved